mongoexport/json.go

```diff
@@ -89,27 +87,26 @@ func (jsonExporter *JSONExportOutput) ExportDocument(document bson.D) error {
 				jsonExporter.Out.Write([]byte("\n"))
 			}
 		}
-		extendedDoc, err := bsonutil.ConvertBSONValueToLegacyExtJSON(document)
+
+		jsonOut, err := bson.MarshalExtJSON(document, jsonExporter.JSONFormat == canonical, false)
 		if err != nil {
 			return err
 		}
-		jsonOut, err := json.Marshal(extendedDoc)
-		if err != nil {
-			return fmt.Errorf("error converting BSON to extended JSON: %v", err)
-		}
+
 		if jsonExporter.PrettyOutput {
 			var jsonFormatted bytes.Buffer
 			json.Indent(&jsonFormatted, jsonOut, "", "\t")
 			jsonOut = jsonFormatted.Bytes()
 		}
 		jsonExporter.Out.Write(jsonOut)
 	} else {
-		extendedDoc, err := bsonutil.ConvertBSONValueToLegacyExtJSON(document)
+		extendedDoc, err := bson.MarshalExtJSON(document, jsonExporter.JSONFormat == canonical, false)
 		if err != nil {
 			return err
 		}
-		err = jsonExporter.Encoder.Encode(extendedDoc)
-		if err != nil {
+
+		extendedDoc = append(extendedDoc, '\n')
```

Comment:
Appending a new line was originally handled by the encoder field, which is no longer necessary because bson.MarshalExtJSON returns the exact bytes we need to write out.